library/src/modules/backup/services/database-services/backup-service.interface.ts

```diff
@@ -0,0 +1,6 @@
+import { DatabaseConfig } from "../../models/shaman-backup.config";
+
+export interface IDatabaseService {
+  type: string;
+  getBackup: (dbConfig: DatabaseConfig) => Promise<string>;
```

Comment:
since the concretions of this are now injected, you probably don't need to pass this as a method parameter, instead should just be available to concretions via construction.

Reply:
Also, since this returns a string, we should probably update the name to indicate the difference (for example, getBackFilePath).

Reply:
> since the concretions of this are now injected, you probably don't need to pass this as a method parameter, instead should just be available to concretions via construction.

I believe the method paramether (dbConfig) is necessary.
I could provide the service with all it's possible database configs during composition based on the type property, but this just changes the parameter getBackupFilePath() would require to the dbName (if a user has multiple databases of the same type, the service needs to find the correct config from the configs provided during composition).